cpp/src/groupby/hash/groupby.cu

```diff
@@ -84,9 +94,114 @@ bool constexpr is_hash_aggregation(aggregation::Kind t)
   // return array_contains(hash_aggregations, t);
   return (t == aggregation::SUM) or (t == aggregation::MIN) or (t == aggregation::MAX) or
          (t == aggregation::COUNT_VALID) or (t == aggregation::COUNT_ALL) or
-         (t == aggregation::ARGMIN) or (t == aggregation::ARGMAX);
+         (t == aggregation::ARGMIN) or (t == aggregation::ARGMAX) or (t == aggregation::MEAN) or
+         (t == aggregation::SUM_OF_SQUARES) or (t == aggregation::STD) or
+         (t == aggregation::VARIANCE);
 }
 
+class hash_compound_agg_finalizer final : public cudf::detail::aggregation_finalizer {
+  size_t i;
+  column_view col;
+  cudf::detail::result_cache& sparse_results;
+  cudf::detail::result_cache* dense_results;
+  rmm::device_vector<size_type> const& gather_map;
+  size_type const map_size;
+  rmm::mr::device_memory_resource* mr;
+  cudaStream_t stream;
+
+ public:
+  hash_compound_agg_finalizer(size_t col_idx,
+                              column_view col,
+                              cudf::detail::result_cache& sparse_results,
+                              cudf::detail::result_cache* dense_results,
+                              rmm::device_vector<size_type> const& gather_map,
+                              size_type map_size,
```

Comment:
Why is `map_size` needed? Isn't `gather_map.size()` sufficient?

Reply:
gather_map.size() is not same as `map_size`. 
`extract_populated_keys` returns gather_map, map_size. (and they choose not to resize gather_map. Seem unnecessary to resize because it's not returned, temporary only).

https://github.com/rapidsai/cudf/blob/9e240f29899c157b8a66e0ce487dcdf80a88a799/cpp/src/groupby/hash/groupby.cu#L334-L337

Reply:
I see. This is probably because a `device_vector::resize` invokes a new allocation and copy even when resizing to a smaller size.